### ISSUE TITLE
Roadmap Ahead

### ISSUE THREAD

**Issue by villanuevawill** [Bounties-Network/BountiesAPI#1]
These are some of the areas we need to begin to tackle.  Feedback on prioritization and any other discussion is encouraged.

## New Features
- Events API to track diffs/events in the contract.  This could be used as the groundwork for analyzing datapoints such as user behavior, reputation, recommended pricing, spam, etc.
- Reputation metrics
- A better definition of user profiles
- Leaderboards and categories filtered by platform (gitcoin, bounties network, etc.)
- Recommended pricing
- Spam detection
- Analytics tag for each platform or vertical to include on their application. This tag would enrich our datasets and enable us to provide better metrics around reputation, pricing, etc.. We would also want to include a dashboard for users where they may remove access to their data.
- Websocket endpoint to provide our structured data to clients/platforms after we write it to our DB.

## Cleanup and Organizational Tasks
-  Setup Linters
-  Ops Documentation
-  Contribution Guidelines
-  Circle Integration and automatic deployments
-  Basic Tests

**Comment by mbeylin** [comment]
>A better definition of user profiles
Building on this more definitely, some way for us to authenticate various oAuth accounts of our users (github, twitter, etc) to let people KYC themselves.

It would also be cool for people to be able to have off-chain profiles, including things like a profile picture and a description of themselves.

- I would also like the ability to comment, off-chain, either on a bounty, or on a particular user. Threaded comments would be cool here as well.

- I would like to be able to rate a user or a submission (off-chain)

**Comment by villanuevawill** [comment]
@mbeylin awesome points. the KYC and off-chain profiles sounds solid. I'd ideally like to use someone like uport - but who knows when that will be officially out.  I definitely think brainstorming a profile/image/etc. solution now is the right thing to do.

Any reason why a fulfillment can't have a rating field on it?

Regarding off-chain comments. Would these be publicly viewable or do you think it is worth introducing a concept of private threads as well?

**Comment by mbeylin** [comment]
@villanuevawill Fulfillments are created by the fulfiller, and they are the only individuals who are able to update their fulfillments. Since the rating for a submission is created after it is on-chain, it would be impossible for anyone but the fulfiller to change it.

**Comment by villanuevawill** [comment]
@mbeylin right... in that case the schema wouldn't make sense. We could expose a rating via the api and store it then. As the api is public - I assume we want to keep all of this public?  Also - still curious on private threads? For v1 it would be easiest to keep everything public. thoughts?

**Comment by mbeylin** [comment]
Agreed on keeping all of this public for now. Private bounties are cool but require a nontrivial increase in complexity to encrypt the bounty data (IPFS hash), and manage the sharing of the key(s) for approved fulfillers.
